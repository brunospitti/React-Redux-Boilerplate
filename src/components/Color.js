import React from 'react';
import styled from "react-emotion";

import { Button } from './Button';


class Color extends React.Component {
    render(){
        const { color, changeColor } = this.props;

        const ColoredBg = styled("div")`
            background-color: rgb(${color[0]}, ${color[1]}, ${color[2]});
        `;

        return(
            <div className="color-wrapper">
                <ColoredBg>
                    <div className="color-rgb">
                        RGB( {color.map( (colorU, i) => <span key={i}>{ colorU }{ (color.length !== i + 1) && ', ' }</span> )} )
                    </div>
                </ColoredBg>
                <Button secondary text="Change color" clickBehav={changeColor(color)}/>
            </div>
        )
    }
};

export default Color;