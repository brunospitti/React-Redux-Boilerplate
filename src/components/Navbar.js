import React from "react";
import { Link } from "@reach/router";
import styled from "styled-components";

import { colors } from '../globalStyles';


// styles with styled-components
const Container = styled("header")`
  background-color: ${colors.primary};
  position: sticky;
  top: 0;
  z-index: 10;
  padding: 20px;
`;

const NavLink = styled(Link)`
  font-size: 20px;
  color: white;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`;

export class Navbar extends React.PureComponent {

  render() {
    return (
      <Container>
        <NavLink to="/">React + Redux</NavLink>
        <NavLink to="/search-params">
        </NavLink>
      </Container>
    );
  }
}
