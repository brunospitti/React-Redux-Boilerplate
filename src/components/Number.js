import React from 'react';

class Number extends React.Component {
    render(){
        const {num, addNum} = this.props;
        return(
            <div className="number-wrapper">
                <p>
                    <strong>Times the button has been clicked:</strong> {num}
                </p>
                <button
                    className="btn increment-number"
                    onClick={() => addNum(num) }
                >
                    Increment number
                </button>
            </div>
        )
    }
};

export default Number;