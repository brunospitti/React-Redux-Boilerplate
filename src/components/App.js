import React from "react";
import { render } from "react-dom";
import { Router } from "@reach/router";
import Loadable from "react-loadable";
import { Provider } from "react-redux";

import store from "../redux/store";
import { Navbar } from "./Navbar";

// global css styles
// eslint-disable-next-line
import globalStyles from '../globalStyles'

// Code splitting section

const loading = () => <h1>loading split code...</h1>;

const LoadableHomePage = Loadable({
  loader: () => import("../pages/HomePage"),
  loading
});

class App extends React.PureComponent {
  render() {
    return (
      <div>
        <Navbar />
        <Provider store={store}>
          <Router>
            <LoadableHomePage path="/" />
          </Router>
        </Provider>
      </div>
    );
  }
}

render(<App />, document.getElementById("root"));
