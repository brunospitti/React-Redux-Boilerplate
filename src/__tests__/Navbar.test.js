import React from "react";
import { create } from "react-test-renderer";
import { Navbar } from "../components/Navbar";

test("snapshot", () => {
  const c = create(<Navbar />);
  expect(c.toJSON()).toMatchSnapshot();
});