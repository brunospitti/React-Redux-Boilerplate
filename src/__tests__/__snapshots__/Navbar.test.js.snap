// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<header
  className="css-1co4p4f"
  onClick={[Function]}
>
  <a
    aria-current="page"
    className="css-etqpg5"
    href="/"
    onClick={[Function]}
  >
    React + Redux
  </a>
  <a
    className="css-etqpg5"
    href="/search-params"
    onClick={[Function]}
  />
</header>
`;
